fedora_messaging/_session.py

```diff
@@ -180,13 +180,18 @@ def _connect_and_publish(self, exchange, message):
             if self._confirms:
                 self._channel.confirm_delivery()
 
+        routing_key = message._encoded_routing_key
+        if config.conf["topic_prefix"]:
```

Comment:
Did you find adding this in the Message class to be a problem? I was thinking setting it on message creation so inspection of the message object would show what's actually sent, but I've not finished my coffee yet and maybe I'm not taking something into account.

Reply:
I have tried both, I think each could work, but then I'd rename `_encoded_routing_key` to something like `_final_routing_key` or `_wire_routing_key`, since it's not only an encoding. Any thoughts on that?

Reply:
Is mutating Message.topic instance variable a problem? Then you could just do ``self.topic = ...`` in ``__init__`` and ``_encoded_routing_key`` would have the right value. In other words, the Message init would look like:

```
def __init__(topic=None, ...):
    if topic:
        # Default is "" on base class
        self.topic = topic

    if config.conf["topic_prefix"]:
        self.topic = ".".join((config.conf["topic_prefix"].rstrip("."), self.topic))
```

I know it's a _bit_ unexpected (I set topic to x, why is it y.x!?), but only if it is explicitly configured.

Reply:
Yeah, I thought about that and figured it would feel weird to have an instance attribute with the same name as a constructor argument but with a slightly different value. But yeah, maybe that's just me, I'll do that.